packages/strapi-connector-bookshelf/lib/buildQuery.js

```diff
@@ -79,43 +79,50 @@ const buildJoinsAndFilter = (qb, model, whereClauses) => {
    * @param {Object} destinationInfo - destination with which we are making a join
    */
   const buildJoin = (qb, assoc, originInfo, destinationInfo) => {
-    if (assoc.nature === 'manyToMany') {
+    if (['manyToMany', 'manyWay'].includes(assoc.nature)) {
       const joinTableAlias = generateAlias(assoc.tableCollectionName);
 
+      let originColumnNameInJoinTable = `${joinTableAlias}.`;
+      if (assoc.nature === 'manyToMany') {
+        originColumnNameInJoinTable += `${singular(
```

Comment:
I would avoid the concatenation here and leave the joinTableAlias in the templateString.